modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientSqlUtils.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2021 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.platform.client;
+
+import org.apache.ignite.binary.BinaryObjectException;
+import org.apache.ignite.internal.binary.BinaryRawReaderEx;
+import org.apache.ignite.internal.binary.BinaryRawWriterEx;
+import org.apache.ignite.internal.binary.BinaryReaderExImpl;
+import org.apache.ignite.internal.binary.BinaryUtils;
+import org.apache.ignite.internal.binary.GridBinaryMarshaller;
+import org.apache.ignite.internal.processors.odbc.SqlListenerUtils;
+import org.jetbrains.annotations.Nullable;
+
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.TimeZone;
+
+/**
+ * SQL-related utility methods for thin client.
+ */
+public class ClientSqlUtils {
+    /**
+     * Read arguments for SQL query.
+     *
+     * @param reader Reader.
+     * @param protoCtx Protocol context.
+     * @return Arguments.
+     */
+    @Nullable public static Object[] readQueryArgs(BinaryRawReaderEx reader, ClientProtocolContext protoCtx) {
+        int cnt = reader.readInt();
+
+        if (cnt > 0) {
+            Object[] args = new Object[cnt];
+
+            for (int i = 0; i < cnt; i++)
+                args[i] = readSqlField(reader, protoCtx);
+
+            return args;
+        }
+        else
+            return null;
+    }
+
+    /**
+     * @param reader Reader.
+     * @param protoCtx Protocol context.
+     * @return Read object.
+     * @throws BinaryObjectException On error.
+     */
+    @Nullable
+    public static Object readSqlField(BinaryRawReaderEx reader, ClientProtocolContext protoCtx)
+            throws BinaryObjectException {
+        BinaryReaderExImpl reader0 = (BinaryReaderExImpl) reader;
+        TimeZone serverTz = TimeZone.getDefault();
+        TimeZone clientTz = protoCtx.clientTimeZone();
+
+        byte type = reader0.readByte();
+        switch (type) {
+            case GridBinaryMarshaller.DATE: {
```

Comment:
Redundant `{` / `}` as per code style, here and below

Reply:
Actually, according to guideline the only case when curly braces are not needed is when the following expression is one-liner (there we have 2 lines of code in every code block)